### PR TITLE
Fix mypy errors and force it on CI

### PR DESCRIPTION
**Pull request by yeputons**

### CODE REVIEW

**Comment by yeputons** [comment]
Just updated from the `main` branch.

File: test_runner/Pipfile [hunk 1]
```diff
@@ -15,10 +15,13 @@ cached-property = "*"
 
 [dev-packages]
 flake8 = "*"
-mypy = "*"
 # Behavior may change slightly between versions. These are run continuously,
 # so we pin exact versions to avoid suprising breaks. Update if comfortable.
 yapf = "==0.31.0"
+mypy = "==0.910"
+# Non-pinned packages follow.
```
Comment (MMeent) [inline test_runner/Pipfile:22]: flake8 is not pinned to a version either, could you make that more consistent?

**Review by kelvich — APPROVED** [review]

File: test_runner/fixtures/zenith_fixtures.py [hunk 2]
```diff
@@ -693,9 +705,9 @@ def stop_and_destroy(self) -> 'Postgres':
         """
 
         assert self.node_name is not None
-        assert self.tenant_id is not None
```
Comment (MMeent) [inline test_runner/fixtures/zenith_fixtures.py:696]: Why remove the assertion if that should always be the case here?

**Review by MMeent — CHANGES_REQUESTED** [review]
Overall OK with the changes, some small changes comments added

File: test_runner/Pipfile (hunk 1 above)
Comment (yeputons) [reply test_runner/Pipfile:22]: However, if someone is to run `flake8` and check their code for warnings, I think running the most recent version is better than an arbitrary pinned one.

**Pull request merged**